modules/oadp-using-enable-api-group-versions.adoc

```diff
@@ -0,0 +1,44 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/advanced-topics.adoc
+
+
+:_content-type: PROCEDURE
+[id="oadp-using-enable-api-group-versions_{context}"]
+= Working with different Kubernetes API versions in the same OADP operation
+
+By default, Velero only backs up resources that use the `preferredAPI` Kubernetes version. However, Velero also includes a feature, link:https://velero.io/docs/v1.9/enable-api-group-versions-feature/[Enable API Group Versions], that overcomes this limitation. When enabled on the source cluster, this feature causes Velero to back up _all_ Kubernetes API group versions that are supported on the cluster, not only the preferred one.
+
+[NOTE]
+====
+As of August 2022, Enable API Group Versions is still in beta.
+====
+
+For example, consider the resource `horizontalpodautoscalers`, which falls under the `autoscaling` group and has 3 API versions - `v1`,`v2beta1`, and `v2beta2` - registered on the cluster, with the preferred API being `v1`.
+
+Without the Enable API Group Versions enabled, Velero backs up only the preferred API group version for the `autoscaling` group, `v1`. With the feature enabled, Velero backs up the remaining supported versions, `v2beta1` and `v2beta2`. Once the versions are stored in the backup tarball file, they are available to be restored on the destination cluster.
+
+When the Enable API Group Versions feature is enabled on the destination cluster, Velero selects the version to restore on the basis of the order of priority of API group versions.
+
+Velero uses the following algorithm to assign priorities to API versions, with 1 as the top priority:
+
+Priority 1: Preferred version of the _destination_ cluster+
+Priority 2: Preferred version of the source_ cluster  +
+Priority 3: Common non-preferred supported version with the highest Kubernetes version priority
+
+.Procedure
+
+* Configure the `EnableAPIGroupVersions` feature flag:
```

Comment:
Configure this _where_? What is this `spec` related to? In what API is this from? Every API in OCP almost without exception has a `spec` field, so this by itself doesn't help me understand where this takes place.

Reply:
@kaovilai @dymurray -- can you help here?

Reply:
Configure right here.. this spec is part of DataProtectionApplication spec
https://github.dev/RichardHoch/openshift-docs/blob/9f03e415c9fe53cdbfcfe095eb482cf1f0dbb05c/modules/oadp-using-enable-api-group-versions.adoc#L35-L39

Dpa spec is first described here
https://github.dev/RichardHoch/openshift-docs/blob/9f03e415c9fe53cdbfcfe095eb482cf1f0dbb05c/modules/oadp-installing-dpa.adoc#L41-L80

Reply:
This API is part of the feature flags parameter for velero which is an option you can configure in DPA.

Reply:
@kaovilai 

So woud the full yaml would be as follows?

apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
----
----
----
spec:
  configuration:
    velero:
      featureFlags:
      - EnableAPIGroupVersions

Reply:
@kaovilai, do we know what namespace this belongs in when it's created, and what the proper `name` is for the API object? Is it user-supplied or is there an canonical name like for `config.openshift.io` APIs? Thanks!